docs/src/tutorial.jl

```diff
@@ -708,161 +707,87 @@ end
 end
 
-# and give `dispatch_step!` to the `agent_step!` keyword during model creation.
-
-# ## Multiple agent types with `@multiagent`
-
-# [`@multiagent`](@ref) is a macro, and hence not "native Julia" syntax like `Union`,
-# however it has been designed to be as similar to [`@agent`](@ref) as possible.
-# The syntax to use it is like so:
-
-@multiagent struct MultiSchelling{X}(GridAgent{2})
-    @subagent struct Civilian # can't re-define existing `Schelling` name
-        mood::Bool = false
-        group::Int
-    end
-    @subagent struct Governor{X} # can't redefine existing `Politician` name
-        group::Int
-        influence::X
-    end
-end
-
-# This macro created three names into scope:
-
-(MultiSchelling, Civilian, Governor)
-
-# however, only one of these names is an actual Julia type:
-
-fieldnames(MultiSchelling)
-
-# that contains all fields of all subtypes without duplication, while
+# and then passing
 
-fieldnames(Civilian)
+model = StandardABM(
+    Union{Schelling, Politician}, # type of agents
+    space; # space they live in
+    agent_step!
+)
 
-# doesn't have any fields. Instead,
-# you should think of `Civilian` and `Governor` as just convenience functions that have been
-# defined for you to "behave like" types. That's why we call these **kinds** and not
-# **types**.
+# ## Multiple agent types with `@sumtype`
 
-# E.g., you can initialize
+# By using `@sumtype` from `DynamicSumTypes.jl` it is possible to improve the 
+# computational performance of simulations requiring multiple types, while almost
+# everything works the same 
 
-civ = Civilian(; id = 2, pos = (2, 2), group = 2) # default `mood`
+using DynamicSumTypes
 
-# or
+@sumtype MultiSchelling(Schelling, Politician) <: AbstractAgent
 
-gov = Governor(; id = 3 , pos = (2, 2), group = 2, influence = 0.5)
+# Now when you create instances you will need to enclose them in `MultiSchelling`
 
-# exactly as if these were types made with [`@agent`](@ref).
-# These are all of type `MultiSchelling`
+p = MultiSchelling(Politician(model; pos = random_position(model), preferred_demographic = 1))
 
-typeof(gov)
+# agents are then all of type `MultiSchelling`
 
-# and hence you can't use `typeof` to differentiate them. But you can use
+typeof(p)
 
-kindof(gov)
+# and hence you can't use only `typeof` to differentiate them. But you can use
 
-# instead.
+typeof(variant(p))
 
-# Since these kinds are not truly different Julia types, multiple dispatch cannot
-# be used to create different agent stepping functions for them.
-# The simplest "native Julia" solution here is to create a function where `if`
-# clauses decide what to do:
+# instead. Hence, the agent stepping function should become something like
 
-function multi_step!(agent, model)
-    if kindof(agent) == :Civilian
-        civilian_step!(agent, model)
-    elseif kindof(agent) == :Governor
-        politician_step!(agent, model)
-    end
-end
+agent_step!(agent, model) = agent_step!(agent, model, variant(agent))
```

Comment:
Same comment as above. We got rid of the `if` clause the dispatches using `kindof`, but we now fundamentally adjust the agent stepping function to have 3 input arguments. This is not harmonious with the rest of the library. I feel like so far this new approach *decreases* the overall harmony between the three scenarios: 1 agent, Union agent, and our macro/sumtype agent.

Reply:
actually you can even improve speed by utilize the variant sometimes, because if you modify the variant you modify the type, but faster. But this is a bit more advanced and so I didn't mention it :D 

But consider that anyway it doesn't require a new macro (`@dispatch`) and inside a module it doesn't require also the other (`@finalize_dispatch`) and the messy thing when you want to use the function before the end of the module

Reply:
This is another point, this is actually almost plain and simple. It doesn't require the fair bit of DSL we needed for `@multiagent`, which really increased more complexity in my opinion